### ISSUE TITLE
Spectator tutorial fails to get key names from its team (works if global)

### ISSUE THREAD

**Issue by illwieckz** [Unvanquished/Unvanquished#1187]
See the "follow" bind advice:

[![spectator bind](https://dl.illwieckz.net/b/unvanquished/bugs/spectator-bind/unvanquished_2020-04-24_211753_000.jpg)](https://dl.illwieckz.net/b/unvanquished/bugs/spectator-bind/unvanquished_2020-04-24_211753_000.jpg)

Note that even as spectator it is displayed correctly when following an human or an alien:

[![spectator bind](https://dl.illwieckz.net/b/unvanquished/bugs/spectator-bind/unvanquished_2020-04-24_211814_000.jpg)](https://dl.illwieckz.net/b/unvanquished/bugs/spectator-bind/unvanquished_2020-04-24_211814_000.jpg)

This bind works in all case:

```
bind  MOUSE3        +attack3
```

This reproduces the bug:

```
teambind aliens     MOUSE3 +attack3
teambind humans     MOUSE3 itemact grenade
teambind spectators MOUSE3 +attack3
```

This issue is blocking for `0.52`.

**Comment by slipher** [comment]
When this line executes in CG_TutorialText

    CG_GetBindings( (team_t) ps->persistant[ PERS_TEAM ] );

What argument is it passing into CG_GetBindings?

**Comment by illwieckz** [comment]
Probably `TEAM_NONE`…

**Comment by illwieckz** [comment]
I added:

```c++
	Log::Warn("TEAM: %d", team);
```

to `CG_GetBindings()` and it prints me `Warn: TEAM: 0` continuously, `TEAM_NONE` seems to be `0`:

https://github.com/Unvanquished/Unvanquished/blob/f79ea6e9b10f6d0d0c32b59f9ed371e099c52b9a/src/shared/bg_public.h#L48-L57

**Comment by illwieckz** [comment]
Interesting:

https://github.com/DaemonEngine/Daemon/blob/393e1829d5acffc0f0fde14c87fd8dd9e7141445/src/engine/client/key_binding.cpp#L384-L385

```c++
// Assumes 'three' teams: spectators, aliens, humans
static const char *const teamName[] = { "default", "aliens", "humans", "spectators" };
```

https://github.com/DaemonEngine/Daemon/blob/393e1829d5acffc0f0fde14c87fd8dd9e7141445/src/engine/client/key_binding.cpp#L213

```c++
static BindTeam bindTeam = BIND_TEAM_SPECTATORS; // Should never be BIND_TEAM_DEFAULT
```

**Comment by illwieckz** [comment]
Note that if we rewrite this thing, I would recommend:

```c++
static const char *const teamName[] = { "default", "spectators", "aliens", "humans",  };
```

So it would be possible to extend this way:

```c++
static const char *const teamName[] = { "default", "spectators", "aliens", "humans",
	"yokai", "predators", "replicators", "terminators", "casper" };
```

**Comment by slipher** [comment]
Yeah so, you want it to be 3 for spectators.

**Comment by illwieckz** [comment]
Better fix the team ordering in both side while we are at it, no?

**Comment by slipher** [comment]
Nah, it will probably break 1000 things if you try to change the numbering.
